### PR TITLE
Enforce linting in CI for C++ and CMake files

### PR DESCRIPTION
**Pull request by schnellerhase**
Changes to CI:
- `ruff` combined to a single call
- `clang-format` combined to a single call and no longer allowed to fail
- `cmake-format` no longer allowed to fail

Additionally this contains the necessary format changes that were previously missed due to allowed failure.

### CODE REVIEW

File: .github/workflows/ccpp.yml [hunk 1]
```diff
@@ -38,40 +38,25 @@ jobs:
...
-          ruff format --check .
-      - name: ruff Python interface checks
-        run: |
-          cd python/
-          ruff check .
-          ruff format --check .
-      - name: mypy checks
+
+      - name: Run ruff
+        run: ruff check python/ cpp/
```
Comment (jhale) [inline .github/workflows/ccpp.yml:46]: The previous setup was deliberate. It checked formatting, and additionally the split allows the user to quickly see in the CI log which part of the Python interface is failing.

File: .github/workflows/ccpp.yml [hunk 2]
```diff
@@ -38,40 +38,25 @@ jobs:
...
-        run: |
-          cd python/dolfinx/wrappers
-          clang-format --version
-          find . -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
-      - name: cmake-format (non-blocking)
-        continue-on-error: true
-        run: |
-          find . -type f \( -name "*.cmake" -o -name "*.cmake.in" -o -name "CMakeLists.txt" \) | xargs cmake-format --check
+
+      - name: Run clang-format
```
Comment (jhale) [inline .github/workflows/ccpp.yml:55]: Non-blocking was intentional.

File: .github/workflows/ccpp.yml [hunk 3]
```diff
@@ -38,40 +38,25 @@ jobs:
...
+        run: find cpp/ python/dolfinx/wrappers/ -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
+
+      - name: Run cmake-format
```
Comment (jhale) [inline .github/workflows/ccpp.yml:58]: Non-blocking was intentional. Both make-format and clang-format are not very stable and behaviour differs widely between versions. This makes it difficult for developers to pass these checks. We periodically do a PR which runs these utilities using the versions in the container images.

File: .github/workflows/ccpp.yml [hunk 4]
```diff
@@ -38,40 +38,25 @@ jobs:
...
-          cd python/dolfinx/wrappers
-          clang-format --version
-          find . -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
-      - name: cmake-format (non-blocking)
-        continue-on-error: true
-        run: |
-          find . -type f \( -name "*.cmake" -o -name "*.cmake.in" -o -name "CMakeLists.txt" \) | xargs cmake-format --check
+
+      - name: Run clang-format
+        run: find cpp/ python/dolfinx/wrappers/ -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
```
Comment (jhale) [inline .github/workflows/ccpp.yml:56]: Split into two steps is intended so it is clear which part of the library is the issue.

**Review by jhale — CHANGES_REQUESTED** [review]

**Comment by schnellerhase** [comment]
In the case of all this being done by intent, I will close the PR.

**Pull request closed without merging**